### PR TITLE
Support for DER encoded keys

### LINKED ISSUE #276: Support querying by asn.1 DER-encoded keys in rest-api
**Problem**
The SDK outputs new keys as asn.1 DER-encoded (ed25519) keys.

The rest-api currently allows querying by the raw key and displays the raw key, but doesn't allow querying by the asn.1/der-encoded key - which has a 6 byte (12 hex character) constant header (302...).

**Solution**
The rest-api should detect that the requested public key input parameter (on all queries) is this asn.1/der-encoded key (based on the length and header), and find the target accounts correctly based on that.

### PR DESCRIPTION
**Pull request by atul-hedera**
Signed-off-by: Atul Mahamuni <atulmahamuni@hedera.com>
**Detailed description**:
Added support for decoding DER encoded keys

**Which issue(s) this PR fixes**:
Fixes #276 

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: rest-api/ed25519.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        let buf = new Uint8Array(Buffer.from(der, 'hex')).buffer;
+        let asn = asn1js.fromBER(buf);
+        if (asn.offset === -1) {
+            return (null); // Not a valid DER/BER format
+        }
+        let asn1Result = asn.result.toJSON(); 
+
+        // Check if it is a ED25519 key
+        const valueBlock = asn1Result.valueBlock.value[0].valueBlock.value[0].valueBlock;
+        if (valueBlock.blockName = 'ObjectIdentifierValueBlock' &&
```
Comment (mike-burrage-hedera) [inline rest-api/ed25519.js]: ==

File: rest-api/ed25519.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    try {
+        let buf = new Uint8Array(Buffer.from(der, 'hex')).buffer;
+        let asn = asn1js.fromBER(buf);
+        if (asn.offset === -1) {
+            return (null); // Not a valid DER/BER format
+        }
+        let asn1Result = asn.result.toJSON(); 
+
+        // Check if it is a ED25519 key
+        const valueBlock = asn1Result.valueBlock.value[0].valueBlock.value[0].valueBlock;
```
Comment (mike-burrage-hedera) [inline rest-api/ed25519.js]: Guaranteed to get a valueBlock of length at least 1?

File: rest-api/ed25519.js (hunk 1 above)
Comment (atul-hedera) [reply rest-api/ed25519.js]: Ouch!

File: rest-api/ed25519.js (hunk 2 above)
Comment (atul-hedera) [reply rest-api/ed25519.js]: I thought the outer try/catch will catch it. But as a safeguard, I'll add the check.

Comment (atul-hedera) [reply rest-api/ed25519.js]: Fixed

File: rest-api/ed25519.js (hunk 1 above)
Comment (atul-hedera) [reply rest-api/ed25519.js]: Fixed

**Pull request merged**